{
  "_args": [
    [
      "require-stack@https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
      "/media/tengex/Storage1/Egyetem/16-17-1/ALKG/alkfejl"
    ]
  ],
  "_from": "require-stack@>=1.0.1 <2.0.0",
  "_id": "require-stack@1.0.2",
  "_inCache": true,
  "_location": "/require-stack",
  "_phantomChildren": {},
  "_requested": {
    "name": "require-stack",
    "raw": "require-stack@https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
    "rawSpec": "https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
    "scope": null,
    "spec": "https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
    "type": "remote"
  },
  "_requiredBy": [
    "/adonis-fold"
  ],
  "_resolved": "https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
  "_shasum": "e00ee348bf96cb5c3e2d4c27b49e41476e08965d",
  "_shrinkwrap": null,
  "_spec": "require-stack@https://registry.npmjs.org/require-stack/-/require-stack-1.0.2.tgz",
  "_where": "/media/tengex/Storage1/Egyetem/16-17-1/ALKG/alkfejl",
  "author": {
    "name": "amanvirk"
  },
  "dependencies": {
    "syntax-error": "^1.1.4"
  },
  "description": "Returns syntax error while requiring modules under try/catch, which is missing by default",
  "devDependencies": {
    "chai": "^3.3.0",
    "istanbul": "^0.3.22",
    "mocha": "^2.3.3",
    "ouch": "^0.2.0-beta",
    "standard": "^6.0.8"
  },
  "keywords": [
    "require-error-trace",
    "require-stack",
    "syntax-error"
  ],
  "license": "MIT",
  "main": "index.js",
  "name": "require-stack",
  "optionalDependencies": {},
  "readme": "# Require Stack\n\n![](https://img.shields.io/travis/thetutlage/require-stack.svg)\n\nNodejs require method does not provide filename or line number for syntax errors when requiring modules under try/catch statement.\nIdeally it is not the problem with Node Js, it has something to do with v8 itself.\n\n`Require Stack` is a wrapper around `require` method, which makes it easier to get syntax error with proper error stack when wrapping require call under `try/catch` block.\n\n## Using Node Js\n\n```javascript\ntry{\n  require('./module/with/syntax/error')\n}catch(e){\n  console.log(e.stack)\n}\n```\n\nabove prints\n\n```\nSyntaxError: Unexpected token {\n    at exports.runInThisContext (vm.js:53:16)\n    at Module._compile (module.js:413:25)\n    at Object.Module._extensions..js (module.js:452:10)\n    at Module.load (module.js:355:32)\n    at Function.Module._load (module.js:310:12)\n    at Module.require (module.js:365:17)\n    at require (module.js:384:17)\n    at Object.<anonymous> (/Users/harmindervirk/workspace/personal/active-packages/require-stack/index.js:11:3)\n    at Module._compile (module.js:434:26)\n    at Object.Module._extensions..js (module.js:452:10)\n```\n\nWhich is no reference to the filename or linenumber where syntax error has occured. File reference on line number 9 is the file from where code is executed.\n\n## Using require stack \n\n```\nvar requireStack = require('require-stack')\ntry{\n  requireStack('./module/with/syntax/error')\n}catch(e){\n  console.log(e.stack)\n}\n```\n\nabove prints\n\n```\n/Users/harmindervirk/workspace/personal/active-packages/require-stack/test/modules/index.js:2\n  return {\n         ^\nParseError: Unexpected token\n```\n\nWhich has reference to the filename with exact line number\n\n## Bonus\n\nIt does all the hard work required to setup error stack which can be parsed by [https://www.npmjs.com/package/stack-trace](https://www.npmjs.com/package/stack-trace).\n\n\n![](http://s9.postimg.org/3xg4vpcbz/Screen_Shot_2015_10_09_at_2_27_07_PM.png)",
  "readmeFilename": "readme.md",
  "scripts": {
    "standard": "standard src/*.js",
    "test": "mocha test --reporter spec --check-leaks --bail"
  },
  "version": "1.0.2"
}
